### ISSUE TITLE
Rewrite in Ocaml/Rust/Zig

### ISSUE THREAD

**Issue by makedonskyi** [diia-open-source/be-user-service#7]
you must do this, because Rust and Zig is memory-safe language

**Comment by Werenter** [comment]
Memory-safety languages are not needed

**Comment by Yusk03** [comment]
> Memory-safety languages are not needed

So time choose Perl!

**Comment by Equilibriumty** [comment]
> Memory-safety languages are not needed

yes they are, memory-safety languages are recommended by the US:

https://www.nsa.gov/Press-Room/Press-Releases-Statements/Press-Release-View/Article/3608324/us-and-international-partners-issue-recommendations-to-secure-software-products/

**Comment by CyberDogFK** [comment]
> https://www.nsa.gov/Press-Room/Press-Releases-Statements/Press-Release-View/Article/3608324/us-and-international-partners-issue-recommendations-to-secure-software-products/

Lol. In the full report is really no JS and TS. JUST: C#, Go, Java, Python, Rust, Swift.

**Comment by Equilibriumty** [comment]
> > https://www.nsa.gov/Press-Room/Press-Releases-Statements/Press-Release-View/Article/3608324/us-and-international-partners-issue-recommendations-to-secure-software-products/
> 
> Lol. In the full report is really no JS and TS. JUST: C#, Go, Java, Python, Rust, Swift.

there is no JS or TS mentioned in the title of this PR

**Comment by Equilibriumty** [comment]
> > > > https://www.nsa.gov/Press-Room/Press-Releases-Statements/Press-Release-View/Article/3608324/us-and-international-partners-issue-recommendations-to-secure-software-products/
> > > 
> > > 
> > > Lol. In the full report is really no JS and TS. JUST: C#, Go, Java, Python, Rust, Swift.
> > 
> > 
> > there is no JS or TS mentioned in the title of this PR
> 
> It's PR to service fully written in TS. So its service may be unsafe and MUST be rewritten🫢

I was replying to this comment

> Memory-safety languages are not needed

**Comment by vtopc** [comment]
![image](https://github.com/diia-open-source/be-user-service/assets/32271530/2dec0f77-6fdb-4513-959e-543e0cc6bf54)
https://github.com/diia-open-source/be-user-service/issues/1
